src/ethereum/arrow_glacier/transactions.py

```diff
@@ -141,14 +141,20 @@ def validate_transaction(tx: Transaction) -> bool:
 
     Returns
     -------
-    verified : `bool`
-        True if the transaction can be executed, or False otherwise.
+    intrinsic_gas : `ethereum.base_types.Uint`
+        The intrinsic cost of the transaction.
+
+    Raises
+    ------
+    InvalidBlock :
+        If the transaction is not valid.
     """
-    if calculate_intrinsic_cost(tx) > Uint(tx.gas):
-        return False
-    if tx.nonce >= U256(U64.MAX_VALUE):
-        return False
-    return True
+    intrinsic_gas = calculate_intrinsic_cost(tx)
+    if intrinsic_gas > tx.gas:
+        raise InvalidBlock
```

Comment:
Should these raise `InvalidTransaction`, since the checks only require the transaction itself?